.github/workflows/audit_build_verify.yml

```diff
@@ -40,72 +40,70 @@ jobs:
-      continue-on-error: true
-
-    - name: Cache next.js build cache
-      uses: actions/cache@3b70782066bd817837b6bd8c518d45b54d91bf76 # pin@v2
-      with:
-        path: .next/cache
-        key: ${{ runner.os }}-next.js-${{ hashFiles('**/package-lock.json') }}
-        restore-keys: |
-          ${{ runner.os }}-next.js-
-      continue-on-error: true
-
-    - name: Build
-      run: npm run build
-
-    - name: Compress build
-      run: tar -zcvf build.tar.gz .next
-
-    - name: Archive build
-      if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/dev' || github.event_name == 'pull_request'
-      uses: actions/upload-artifact@ee69f02b3dfdecd58bb31b4d133da38ba6fe3700 # pin@v2
-      with:
-        name: build
-        path: build.tar.gz
+      - name: Checkout
+        uses: actions/checkout@ee5a9d1395988d7c82c67c69571871f5994b94df # pin@v2
+
+      - name: Cache node modules
+        uses: actions/cache@3b70782066bd817837b6bd8c518d45b54d91bf76 # pin@v2
+        with:
+          path: |
+            .npm
+            node_modules
+          key: ${{ runner.os }}-build-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-
+        continue-on-error: true
+
+      - name: Install dependencies
+        run: npm install --no-audit
+
+      - name: Lint
+        # run only on master/dev branch and pull requests
+        if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/dev' || github.event_name == 'pull_request'
+        run: npm run lint:sarif
+
+      - name: Upload lint results
+        # run if lint failed and only on master/dev branch and pull requests
+        if: always() && (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/dev' || github.event_name == 'pull_request')
+        uses: github/codeql-action/upload-sarif@b1e2c9b8bd0550a90e66c967d9795316f9f21910 # pin@codeql-bundle-20201008
+        with:
+          sarif_file: lint-results.sarif
+        continue-on-error: true
+
+      - name: Cache next.js build cache
+        uses: actions/cache@3b70782066bd817837b6bd8c518d45b54d91bf76 # pin@v2
+        with:
+          path: .next/cache
+          key: ${{ runner.os }}-next.js-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-next.js-
+        continue-on-error: true
+
+      - name: Build
+        run: npm run build
+
+      - name: Compress build
+        run: tar -zcvf build.tar.gz .next
+
+      - name: Archive build
+        if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/dev' || github.event_name == 'pull_request'
+        uses: actions/upload-artifact@ee69f02b3dfdecd58bb31b4d133da38ba6fe3700 # pin@v2
+        with:
+          name: build
+          path: build.tar.gz
 
   e2e:
-      needs: build
-      # run only on master/dev branch
-      if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/dev' || github.event_name == 'pull_request'
```

Comment:
e2e test should run only on PR/master/dev branches

Reply:
Gotcha. For some reason, this was being skipped on commits for this PR.